libdrgn/program.c

```diff
@@ -381,9 +416,18 @@ drgn_program_set_core_dump(struct drgn_program *prog, const char *path)
 		prog->file_segments[j].file_size = phdr->p_filesz;
 		prog->file_segments[j].fd = prog->core_fd;
 		prog->file_segments[j].eio_is_fault = false;
+		prog->file_segments[j].map_address = NULL;
+
+		map_address = mmap(NULL, phdr->p_filesz, PROT_READ,
+				MAP_PRIVATE, prog->core_fd, phdr->p_offset);
+		if (map_address && map_address != MAP_FAILED) {
```

Comment:
According to [`mmap(2)`](https://man7.org/linux/man-pages/man2/mmap.2.html), mmap won't return `NULL` unless `MAP_FIXED` is given.

```suggestion
		if (map_address != MAP_FAILED) {
```